backend/src/main/java/ai/verta/modeldb/versioning/blob/container/DatasetContainer.java

```diff
@@ -73,33 +78,39 @@ public void process(
       default:
         throw new ModelDBException("Unknown blob type", Code.INTERNAL);
     }
-    rootTree.push(locationList, blobHash, blobType);
+    if (blobHash != null) {
+      rootTree.push(locationList, blobHash, blobType);
+    }
   }
 
   static String saveBlob(Session session, PathDatasetBlob path, Set<String> blobHashes)
       throws NoSuchAlgorithmException {
     // sorted
     Map<String, AutogenPathDatasetComponentBlob> componentHashes = new LinkedHashMap<>();
-    for (AutogenPathDatasetComponentBlob componentBlob :
-        AutogenPathDatasetBlob.fromProto(path).getComponents()) {
-      final String componentHash = computeSHA(componentBlob);
-      componentHashes.put(componentHash, componentBlob);
-    }
-    String blobHash = computeSHAPathDataset(componentHashes);
-    if (!blobHashes.contains(blobHash)) {
-      blobHashes.add(blobHash);
-      for (Map.Entry<String, AutogenPathDatasetComponentBlob> component :
-          componentHashes.entrySet()) {
-        if (!blobHashes.contains(component.getKey())) {
-          blobHashes.add(component.getKey());
-          PathDatasetComponentBlobEntity pathDatasetComponentBlobEntity =
-              new PathDatasetComponentBlobEntity(
-                  component.getKey(), blobHash, component.getValue().toProto().build());
-          session.saveOrUpdate(pathDatasetComponentBlobEntity);
+    AutogenPathDatasetBlob autogenPathDatasetBlob = AutogenPathDatasetBlob.fromProto(path);
```

Comment:
why was this required?

Reply:
There was a bug when I was testing this feature in case of creating a notebook with an empty path.

I can move it to a separate PR if needed

Reply:
fine for now , but please avoid this in future